lib/solr_configurator.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require 'zip'
+
+class SolrConfigurator
+  def initialize
+    connection
+  end
+
+  def zip_file
+    File.open(tempfile)
+  end
+
+  def configset_exists?
+    resp = @connection.get('/solr/admin/configs?action=LIST&omitHeader=true')
```

Comment:
I would avoid using `@connection` entirely in this class, and instead call the private method `connection`. That way there's no chance of a dependency or race condition between the private method being called first (in this case in the constructor), setting the instance variable, and then subsequent methods calling it. Plus what if in the future you needed to do something fancy in that method beyond just setting a single variable? By using the private method you're basically abstracting the implementation away.

Reply:
doh! the lightbulb just went off. ok. I can certainly fix that.

Reply:
That's a pattern I learned from Sandi Metz 👍

Reply:
Exactly. Your `#initialize` method can go away